msgraph/directory_objects_test.go

```diff
@@ -0,0 +1,190 @@
+package msgraph_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/manicminer/hamilton/auth"
+	"github.com/manicminer/hamilton/internal/test"
+	"github.com/manicminer/hamilton/internal/utils"
+	"github.com/manicminer/hamilton/msgraph"
+	"github.com/manicminer/hamilton/odata"
+)
+
+type DirectoryObjectsClientTest struct {
+	connection   *test.Connection
+	client       *msgraph.DirectoryObjectsClient
+	randomString string
+}
+
+func TestDirectoryObjectsClient(t *testing.T) {
+	rs := test.RandomString()
+	c := DirectoryObjectsClientTest{
+		connection:   test.NewConnection(auth.MsGraph, auth.TokenVersion2),
+		randomString: rs,
+	}
+	c.client = msgraph.NewDirectoryObjectsClient(c.connection.AuthConfig.TenantID)
+	c.client.BaseClient.Authorizer = c.connection.Authorizer
+
+	g := GroupsClientTest{
+		connection:   test.NewConnection(auth.MsGraph, auth.TokenVersion2),
+		randomString: rs,
+	}
+	g.client = msgraph.NewGroupsClient(c.connection.AuthConfig.TenantID)
+	g.client.BaseClient.Authorizer = c.connection.Authorizer
+
+	u := UsersClientTest{
+		connection:   test.NewConnection(auth.MsGraph, auth.TokenVersion2),
+		randomString: rs,
+	}
+	u.client = msgraph.NewUsersClient(c.connection.AuthConfig.TenantID)
+	u.client.BaseClient.Authorizer = c.connection.Authorizer
```

Comment:
it might be worth adding some test-wrappers here?

Reply:
+1